Lib/csv.py

```diff
@@ -447,3 +449,14 @@ def has_header(self, sample):
                     hasHeader -= 1
 
         return hasHeader > 0
+
+
+    def _guess_lineterminator(self, sample):
+        # Guess line terminator based on frequency
+        terminators = {le: sample.count(le) for le in ('\r\n', '\n', '\r')}
```

Comment:
Every '\r\n' will be counted as '\r' and '\n' as well as '\r\n'.  So I believe the '\r\n' count should be subtracted from the other 2 before comparing.